### PR TITLE
chore: move plugins folder outside of Composer

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
## Description

Also renames it to extensions.

This will help avoid yarn workspace issues with transient dependencies. Extensions should be considered external/isolated modules.

## Task Item

#minor

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
Can someone on windows test this with a clean checkout?

**Comment by benbrown** [comment]
This scares me! There is a big PR involving the azurePublish plugin.  Any chance we can let that land before we move all the files around?

**Comment by a-b-r-o-w-n** [comment]
@benbrown git should be able to handle the conflict since no source code was changed.

**Comment by a-b-r-o-w-n** [comment]
@benbrown if you tell me the branch you are working on I can do a merge locally to see.

**Comment by benbrown** [comment]
`wenyluo/provision`

File: Composer/packages/extension/src/extensionContext.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ class ExtensionContext {
     }
   }
 
+  /** @deprecated */
   public async loadPluginsFromFolder(dir: string) {
```
Comment (tonyanziano) [inline Composer/packages/extension/src/extensionContext.ts:107]: Is there anything using this that would benefit from knowing that this is deprecated? If it's only used internally can we just remove it?

File: Composer/scripts/compileExtensions.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  fs.writeFileSync(buildCachePath, JSON.stringify(buildCache, null, 2));
+};
+
+const shouldCompile = (name, lastModified) => {
+  return buildCache[name] ? new Date(buildCache[name]) < lastModified : true;
+};
+
+const compile = (name, extPath) => {
+  const packageJSON = JSON.parse(fs.readFileSync(path.join(extPath, 'package.json')));
+  const hasBuild = packageJSON && packageJSON.scripts && packageJSON.scripts.build;
```
Comment (tonyanziano) [inline Composer/scripts/compileExtensions.js:74]: Just an idea: Would it be worth it to console.warn() here that if you want your extension built you need to define a `build` script in the `package.json`?

**Review by tonyanziano — COMMENTED** [review]
Some minor comments. I'll defer to how the merging of the azure publish branch goes

File: Composer/scripts/compileExtensions.js (hunk 2 above)
Comment (a-b-r-o-w-n) [reply Composer/scripts/compileExtensions.js:74]: Good idea.

File: Composer/packages/extension/src/extensionContext.ts (hunk 1 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/extension/src/extensionContext.ts:107]: It's used in a single test file. I'll clean it up another time.

File: Composer/scripts/compileExtensions.js [hunk 3]
```diff
@@ -73,12 +73,15 @@ const compile = (name, extPath) => {
...
+  execSync('yarn --force --frozen-lockfile', { cwd: extPath, stdio: 'inherit' });
+
   if (hasBuild) {
-    console.log('[%s] compiling', name);
-    console.log('[%s] yarn --force', name);
-    execSync('yarn --force --frozen-lockfile', { cwd: extPath, stdio: 'inherit' });
     console.log('[%s] yarn build', name);
     execSync('yarn build', { cwd: extPath, stdio: 'inherit' });
+  } else {
+    console.log('[%s] no build script found.', name);
```
Comment (tonyanziano) [inline Composer/scripts/compileExtensions.js:84]: 👍 👍 👍

**Comment by a-b-r-o-w-n** [comment]
That branch isn't set to merge until later next week. So we'll merge this when ready and I will manage the conflicts.

**Review by tonyanziano — APPROVED** [review]
Looks good once merge conflicts are fixed

**Review by tonyanziano — APPROVED** [review]
🚢

**Pull request merged**